dags/common/enricher.py

```diff
@@ -0,0 +1,95 @@
+import xml.etree.ElementTree as ET
+from logging import Logger
+from typing import Dict
+
+import backoff
+import requests
+from structlog import get_logger
+
+
+class Enricher(object):
+    def __init__(self) -> None:
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+
+    def _get_schema(self):
+        return "http://repo.scoap3.org/schemas/hep.json"
+
+    @backoff.on_exception(
+        backoff.expo, requests.exceptions.RequestException, max_time=60, max_tries=3
+    )
+    def _get_arxiv_categories(self, arxiv_id=None):
+        if arxiv_id is None:
+            raise ValueError("Arxiv is None.")
+
+        def clean_arxiv(arxiv):
+            if arxiv is None:
+                return None
+
+            return arxiv.split(":")[-1].split("v")[0].split(" ")[0].strip("\"'")
+
+        def get_arxiv_categories_from_response_xml(xml: ET.Element):
+            xml_namespaces = {
+                "arxiv": "http://arxiv.org/schemas/atom",
+                "w3": "http://www.w3.org/2005/Atom",
+            }
+
+            entries = xml.findall("./w3:entry", namespaces=xml_namespaces)
+            if len(entries) != 1:
+                return []
+            entry = entries[0]
+
+            primary_categories = [
+                node.attrib["term"]
+                for node in entry.findall(
+                    "./arxiv:primary_category", namespaces=xml_namespaces
+                )
+            ]
```

Comment:
The following doesn't work?
```suggestion
               entry.findall(
                    "./arxiv:primary_category/@term", namespaces=xml_namespaces
                )
```

do we need the namespaces? you can strip them imo

Reply:
About the attribute : 
```
FAILED tests/units/common/test_enricher.py::test_get_arxiv_categories_arxiv_id - KeyError: '@'
FAILED tests/units/common/test_enricher.py::test_get_arxiv_categories_title - KeyError: '@'
FAILED tests/units/common/test_enricher.py::test_get_arxiv_categories_doi - KeyError: '@'
```

About the namespaces : 
```
FAILED tests/units/common/test_enricher.py::test_get_arxiv_categories_arxiv_id - SyntaxError: prefix 'arxiv' not found in prefix map
FAILED tests/units/common/test_enricher.py::test_get_arxiv_categories_title - SyntaxError: prefix 'arxiv' not found in prefix map
FAILED tests/units/common/test_enricher.py::test_get_arxiv_categories_doi - SyntaxError: prefix 'arxiv' not found in prefix map
```

Reply:
The lib that we're using, `xml.etree`, is a lot more primitive than other libs